### ISSUE TITLE
The exploration setting of PPOAgent

### ISSUE THREAD

**Issue by marooncn** [tensorforce/tensorforce#572]
Hi, I use the default actions_exploration setting for PPOAgent (If I define actions_exploration, it will always output limit values) . But with the training of the agent, the performance of the network is getting worse. I want to know how does the exploration changes and want if I want to use greedy policy after training?
BTW, I also use a DQNAgent to train, the loss becomes bigger and bigger even if I change the batch size and learning rate, any advice? Thank you.

**Comment by AlexKuhnle** [comment]
Hi @marooncn,
Have you played around with exploration values and [decay mechanisms](https://github.com/tensorforce/tensorforce/tree/master/tensorforce/core/explorations), whether it helps to not output limit values? If you want to use the greedy policy, you can use `act(..., deterministic=True)`, and if you do not train anymore (hence you don't need to call `observe()` after `act()`), you can additionally set `act(..., independent=True)`, so that the model doesn't expect a follow-up `observe()`.

**Comment by marooncn** [comment]
Hi @AlexKuhnle ,
I load the latest trained model  which can succeed more than 50% of my task and use **act(..., deterministic=True)** , but the agent is not stable at all, only about 20% of my task can succeed now. 
That's so weird that it becomes more exploratory.

**Comment by AlexKuhnle** [comment]
Hope it helped... :-)

**Comment by marooncn** [comment]
@AlexKuhnle，Yes, it works :-)

**Comment by marooncn** [comment]
@AlexKuhnle ，what the default exploration function of PPOAgent? I just check the code and it likes 'none', does it mean agent just use output of the network as action for continuous output?  but the agent likes keep exploring because when I use **act(..., deterministic=True)** it just behaves the best strategy which is differnent from use **act(..., deterministic=False)**

**Comment by AlexKuhnle** [comment]
Yes, none works too. The action exploration value only configures the degree of "noise" added to output actions, independent of the model. In the case of policy gradient models like PPO (not for Q models), the action at every timestep is also sampled according to the distribution (which itself is conditioned on the network output, but is not the same as the network output), so this is another source of "noise", which could be seen as exploration. Variable noise, finally, would be another way of infusing "noise", but it's turned off by default. In summary, the difference you observe is the effect of sampling vs deterministically choosing the action.

**Comment by marooncn** [comment]
Got it, Thank you.
